### ISSUE TITLE
How do other nextjs apps do testing?

### ISSUE THREAD

**Issue by mlahargou** [iFixit/react-commerce#1032]

**Comment by dhmacs** [comment]
This is Vercel CEO recommendation:
https://twitter.com/rauchg/status/807626710350839808
<img width="595" alt="Screenshot 2022-11-16 at 01 06 14" src="https://user-images.githubusercontent.com/4640135/202050903-0085266b-c087-43f8-b30f-17afa89aadfd.png">

So short answer is ➡️ Cypress (or Playwright if you like it better)

https://nextjs.org/docs/testing

**Comment by ardelato** [comment]
**TL;DR** 
- End-to-End tests with `Cypress`
- Integration & Unit test with `Jest`, `React Testing Library`, and `Mock Service Worker`
- Unit Tests > E2E Tests

--- 

We should prioritize `Unit Testing` with `Jest` over `E2E` tests with `Cypress` since creating and running `Cypress` tests is expensive.

**Unit Testing**
- One unit of code whether that is a `function`, `component`, or `API` route
- Isolate by mocking dependencies
	- This is where `MSW` and `Jest` mocking come in 
- This type should be preferred for writing granular tests

**Integration Testing**
- Check if parent and child components are interacting correctly
- While we could also test if our APIs are interacting with the databases correctly, I think that is outside the scope of getting these lists of tests written #452.

**E2E Testing**
- We should mainly use this to test user flows from beginning to end
- These tests cover critical paths that are necessary for our basic business logic
